mismi-s3/test/Test/IO/Mismi/S3/Commands.hs

```diff
@@ -306,6 +306,19 @@ prop_write_overwrite (UniquePair x y) = withAWS $ \a -> do
   r <- read a
   pure $ r === pure y
 
+prop_sync_overwrite = forAll (elements muppets) $ \m -> withAWS' $ \a b -> do
+  createSmallFiles a m 10
+  syncWithMode OverwriteSync a b 1
+  syncWithMode OverwriteSync a b 1
+  mapM_ (\e -> exists e >>= \e' -> when (e' == False) (throwM . userError $ "Output files do not exist")) (files b m 10)
```

Comment:
Probably not for this code. This had been uplifted from code where the number of files had been very large (too large to keep around). Will update thanks